eng/pipelines/pr-breaking-check.yml

```diff
@@ -0,0 +1,94 @@
+trigger: none
+
+resources:
+  repositories:
+    - repository: azure-rest-api-specs
+      type: github
+      name: Azure/azure-rest-api-specs
+      endpoint: azure
+    - repository: azure-sdk-tools
+      type: github
+      name: Azure/azure-sdk-tools
+      endpoint: azure
+
+stages:
+  - stage: Build_and_Validate
+    jobs:
+      - job: Validate_generated_swagger
+        pool:
+          name: azsdk-pool-mms-ubuntu-2004-general
+          vmImage: ubuntu-20.04
+        steps:
+          # Step 1: Build TypeSpec
+          - checkout: self
+            submodules: true
+          - template: /eng/pipelines/templates/install.yml
+            parameters:
+              nodeVersion: "18.x"
+              workingDirectory: $(Build.SourcesDirectory)/typespec-azure
+          - script: pnpm run build
+            displayName: Build
+            workingDirectory: $(Build.SourcesDirectory)/typespec-azure
+
+          #Step 2: Update swagger by TypeSpec in Step 1
+          - checkout: azure-rest-api-specs
+            fetchDepth: 1
+          - pwsh: |
```

Comment:
We already have this logic for the cadl-ranch e2e test, could we reuse that instead of having to maintain an extra place and deal with powershell